### PR TITLE
[Access] Refactor REST Routes Package

### PR DESCRIPTION
**Pull request by UlyanaAndrukhiv**
Closes: #6615

## Context
This PR introduces a significant restructuring of the `access/rest` folder, enhancing modularity and clarity.

- The new organization separates `http` and `WebSocket` components into `http` and `websockets` directories, simplifying maintenance and readability.
- Common functionality is moved to the `rest/common` package.
- `RouterBuilder`implementation is separated from all routes and moved to  `rest/router`.
- Lists of `http` and `ws` routes are now in `http_routes.go` and `ws_routes.go`, respectively.
- The current `v1/subscribe_events` route is moved into the `websocket/legacy` folder to separate future `WebSocket` implementations.

### CODE REVIEW

File: engine/access/rest/common/base_http_request_handler.go [hunk 1]
```diff
@@ -11,50 +11,50 @@ import (
...
 
 const MaxRequestSize = 2 << 20 // 2MB
 
-// HttpHandler is custom http handler implementing custom handler function.
-// HttpHandler function allows easier handling of errors and responses as it
+// BaseHttpHandler is custom http handler implementing custom handler function.
+// BaseHttpHandler function allows easier handling of errors and responses as it
 // wraps functionality for handling error and responses outside of endpoint handling.
-type HttpHandler struct {
+type BaseHttpHandler struct {
```
[the remark this answers is not in the document]
Comment (UlyanaAndrukhiv) [reply engine/access/rest/common/base_http_request_handler.go]: [cfd4a7b](https://github.com/onflow/flow-go/pull/6616/commits/cfd4a7b7217e4eeb250a4f88530cb8d8fef17d56)

File: engine/access/rest/router/ws_routes.go [hunk 2]
```diff
@@ -0,0 +1,22 @@
+package router
+
+import (
+	"net/http"
+
+	"github.com/onflow/flow-go/engine/access/rest/websockets/legacy"
+	"github.com/onflow/flow-go/engine/access/rest/websockets/legacy/routes"
+)
+
+type wsroute struct {
```
[the remark this answers is not in the document]
Comment (UlyanaAndrukhiv) [reply engine/access/rest/router/ws_routes.go]: [5e30e69](https://github.com/onflow/flow-go/pull/6616/commits/5e30e696cb49b26fe23f8ec950e6e0f259a2d06c)

**Review by peterargue — COMMENTED** [review]
looks good. only comment is about switching to using strings for payload

File: engine/access/handle_irrecoverable_state_test.go [hunk 3]
```diff
@@ -249,7 +248,7 @@ func (suite *IrrecoverableStateTestSuite) TestRestInconsistentNodeState() {
 // optionsForBlocksIdGetOpts returns options for the BlocksApi.BlocksIdGet function.
 func optionsForBlocksIdGetOpts() *restclient.BlocksApiBlocksIdGetOpts {
 	return &restclient.BlocksApiBlocksIdGetOpts{
-		Expand:  optional.NewInterface([]string{routes.ExpandableFieldPayload}),
+		Expand:  optional.NewInterface([]string{"payload"}),
```
[the remark this answers is not in the document]
Comment (UlyanaAndrukhiv) [reply engine/access/handle_irrecoverable_state_test.go]: @peterargue, thanks for the comment! The original reason for this change is now outdated, so I reverted to using the `const` value.
Commit [90cbd8d](https://github.com/onflow/flow-go/pull/6616/commits/90cbd8dd374fcf9f8606f718c3e0de20d72dddfd).

**Review by peterargue — APPROVED** [review]

**Pull request merged**